indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
+                    self._indexedFlowable[p1] = {'text': escape(contrib.title), 'level': 2}
+
+                p2 = Paragraph(speakers, self._styles["conveners"])
                 res.append(p2)
-                if entry == entriesOnDay[-1]: # if it is the last one, we do the page break and remove the previous one.
-                    i=len(res)-1
-                    while i>=0:
+                if entry == entries[-1]:  # if it is the last one, we do the page break and remove the previous one.
+                    i = len(res) - 1
+                    while i >= 0:
                         if isinstance(res[i], PageBreak):
                             del res[i]
                             break
-                        i-=1
+                        i -= 1
                     if self._ttPDFFormat.showNewPagePerSession():
                         res.append(PageBreak())
-            #break
-            elif self._ttPDFFormat.showBreaksAtConfLevel() and \
-                 isinstance(entry,schedule.BreakTimeSchEntry):
-                breakE=entry
-                room=""
-                if breakE.getRoom() is not None:
-                    room=" - %s"%escape(breakE.getRoom().getName())
-                caption="%s"%breakE.getTitle()
-                text="""<u>%s</u>%s (%s-%s)"""%(
-                        escape(caption),room,
-                        escape(breakE.getAdjustedStartDate(self._tz).strftime("%H:%M")),
-                        escape(breakE.getAdjustedEndDate(self._tz).strftime("%H:%M")))
-                p1=Paragraph(text,self._styles["session_title"])
+            # break
+            elif self._ttPDFFormat.showBreaksAtConfLevel() and entry.type == TimetableEntryType.BREAK:
+                break_entry = entry
+                break_ = break_entry.object
+                room = ''
+                if break_.location_data['room_name']:
+                    room = ' - {}'.format(escape(break_.location_data['room_name']))
+
+                text = '<u>{}</u>{} ({}-{})'.format(escape(break_.title), room,
+                                                    format_time(break_entry.start_dt, timezone=self._tz),
+                                                    format_time(break_entry.end_dt, timezone=self._tz))
+
+                p1 = Paragraph(text, self._styles["session_title"])
                 res.append(p1)
+
                 if self._ttPDFFormat.showTitleSessionTOC():
-                    self._indexedFlowable[p1]={"text":escape(breakE.getTitle()), "level":2}
-                if entry == entriesOnDay[-1]: # if it is the last one, we do the page break and remove the previous one.
-                    i=len(res)-1
-                    while i>=0:
+                    self._indexedFlowable[p1] = {'text': escape(break_.title), 'level': 2}
+
+                if entry == entries[-1]:  # if it is the last one, we do the page break and remove the previous one.
+                    i = len(res) - 1
```

Comment:
definitely, however I didn't get much into improving the code since it is legacy and we will get rid of it eventually